### DISCUSSION TITLE
Automapping deprecation in 6.4.*

### DISCUSSION THREAD

**Discussion by justin-oh** [symfony/symfony#58284] [Q&A]
I saw a deprecation warning about automapping for routes and found [this blog post](https://symfony.com/blog/new-in-symfony-7-1-mapped-route-parameters).

I thought about getting ahead of this deprecation for when the times comes to move my sites from 6.4 to 7.4, but in order to do that I have to use the MapEntity attribute.

My question is, will MapEntity eventually be deprecated in favour of fully using "Mapped Route Parameters"? I don't want to have to essentially do this work twice.

**Comment by stof** [comment]
The MapEntity attribute is not only about configuring the mapping. So it won't be deprecated.
And I don't think we will deprecate the explicit mapping feature that it has (it is essentially free to implement, as most of the processing logic is reused between the handling of an explicit mapping in the attribute and the handling of mapped route parameters). Thus, explicit mappings might still be necessary for some weird advanced cases that don't have an equivalent with mapped route parameters. For now, there is no plan to deprecation the explicit mapping feature.
